classes/ActionScheduler_QueueCleaner.php

```diff
@@ -40,16 +40,32 @@ public function __construct( ActionScheduler_Store $store = null, $batch_size =
 	/**
 	 * Default queue cleaner process used by queue runner.
 	 *
-	 * @return void
+	 * @return array
 	 */
 	public function delete_old_actions() {
 		/**
 		 * Filter the minimum scheduled date age for action deletion.
 		 *
 		 * @param int $retention_period Minimum scheduled age in seconds of the actions to be deleted.
 		 */
-		$lifespan = apply_filters( 'action_scheduler_retention_period', $this->month_in_seconds );
-		$cutoff = as_get_datetime_object( $lifespan . ' seconds ago' );
+		$lifespan = (int) apply_filters( 'action_scheduler_retention_period', $this->month_in_seconds );
+
+		try {
+			$cutoff = as_get_datetime_object( $lifespan . ' seconds ago' );
+		} catch ( Exception $e ) {
+			_doing_it_wrong(
```

Comment:
I added this for completeness, in but didn't cover it in our unit tests because it seems tricky to reliably trigger it across different PHP runtimes (the behavior when casting to int from an unsupported type is [undefined](https://www.php.net/manual/en/language.types.integer.php#language.types.integer.casting)).